### ISSUE TITLE
Can't configure BullModule with useClass

### ISSUE THREAD

**Issue by maksimkurb** [fwoelffel/nest-bull#88]
When I configure `BullModule` via `useClass`, my settings are being ignored.
That's because [here](https://github.com/fwoelffel/nest-bull/blob/a67b1a3e2c5335c3a6573408c0281f08d418ef90/lib/bull.module.ts#L33) we are using [createQueueProviders](https://github.com/fwoelffel/nest-bull/blob/a67b1a3e2c5335c3a6573408c0281f08d418ef90/lib/bull.providers.ts#L42-L48) which accepts `BullModuleOptions[]`, but we pass `BullModuleAsyncOptions[]` into it.
`createBullOptions()` from `BullOptionsFactory` is never used in the code.
I have started writing [a fix](https://github.com/maksimkurb/nest-bull/commit/73ff0b83ac4fe1bba3139718ce3cc37447f7e90d) but tests with useClass scenario are failing: mock function is never called.

**Comment by fwoelffel** [comment]
Thanks for reporting this @maksimkurb :1st_place_medal: 
It's nice to see that you prepared the failing test that can confirm your bug report. Could you create the pull request before we continue to work on this?

**Comment by maksimkurb** [comment]
Closing in favor of #89

**Comment by fwoelffel** [comment]
Don't close an issue until it's resolved.

**Comment by fwoelffel** [comment]
I believe this is rather an error in the `BullModuleAsyncOptions` interface:
```ts
export interface BullModuleAsyncOptions extends Pick<ModuleMetadata, 'imports'> {
  name?: string;
  useExisting?: Type<BullOptionsFactory>;
  useClass?: Type<BullOptionsFactory>;
  useFactory?: (
    ...args: any[]
  ) => Promise<BullModuleOptions> | BullModuleOptions;
  inject?: any[];
}
```
should be
```ts
export interface BullModuleAsyncOptions extends Pick<ModuleMetadata, 'imports'> {
  name?: string;
  useExisting?: Type<BullModuleOptions>;
  useClass?: Type<BullModuleOptions>;
  useFactory?: (
    ...args: any[]
  ) => Promise<BullModuleOptions> | BullModuleOptions;
  inject?: any[];
}
```

I'll try to have a look at this by the end of the week.

In the meantine, I think you could refactor your async configuration and use `useFactory` instead of `useClass`.

**Comment by fwoelffel** [comment]
I think I'll have a look at your PR and try to fix the current implementation of the `useClass` alternative without breaking the interface. But keep #94 in mind. I might change this behavior pretty soon.

**Comment by fwoelffel** [comment]
Since this is broken, I've made my mind to directly "fix" this through #94. I'm therefore closing this in favor of #94.
